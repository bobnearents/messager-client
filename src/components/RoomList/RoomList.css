.room-list-container {
  margin:0;
  width:20%;
  height:89.5vh;
  background:#264653;
  display:inline-flex;
  overflow: hidden;
  color:white;
}

.room-list, .dm-list {
  height:32%;
  min-height:150px;
  padding:0;
  display:inline-flex;
  flex-direction: column; 
  width:100%;
  align-items: center;
  overflow-y:scroll;
}


.room-list div {
  margin:8px 0;
  width:100%;
  padding:4px;
  text-align: center;
  font-size: 1.2em;
}

.room-list a {
  width:100%;
  color:#CFDBD5;
  text-decoration:none;
}

.room-list div:hover {
  background:#192F38;
}

.current-room{
  background:#CFDBD5;
}

.current-room:hover {
  background:#CFDBD5;
}

.current-room a {
  color:#264653;
}

body .active {
  display:inline-block;
}

.inactive {
  display:none;
}

.logout {
  color:#CFDBD5;
  text-align: center;
  display:block;
}

.logout:hover {
  text-decoration: none;
}

.add-room {
  color:#CFDBD5;
  vertical-align: middle;
  margin-left:6px;
  font-size: .9em;

  
}

.add-room:hover {
  background:#79C99E;
  border-radius: 10px;
  padding:4px;
  color:black;
  background:#79C99E;
}


@media(max-width:668px) {

  .room-list-container {
    display:none;
    width:100%;
  }
}
