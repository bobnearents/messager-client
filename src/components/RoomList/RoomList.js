import React, { Component } from 'react';
import MessageContext from '../../context/message-context'
import { Link } from 'react-router-dom'
import './RoomList.css'
import { FaPlus } from "react-icons/fa";

class MessageList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  static contextType = MessageContext;

  handleLogoutClick = () => {
    window.localStorage.clear();
    this.context.setUser('newcomer')
  };

  generateRoomsHtml = roomsArray => {
    return roomsArray.map((room, index) => {
      return (
        <div key={index} className={room.id == this.context.room_id ? "current-room" : ""}>
          <Link
            onClick={() => this.context.changeRoom(room.id)}
            to={`/rooms/${room.id}`}
            value={room.id}
          >
            {room.name}
          </Link>
        </div>
      );
    });
  };

  render() {
    return (
      <MessageContext.Consumer>
        {value => {
          return (
            <div
              className={
                value.isRoomActive
                  ? "active room-list-container"
                  : "inactive room-list-container"
              }
            >
              <h2>
                Rooms
                <Link to="/createRoom">
                  <FaPlus className="add-room" />
                </Link>
              </h2>

              <div className="room-list">
                {this.generateRoomsHtml(value.rooms)}
              </div>
              
              <Link className="logout" onClick={this.handleLogoutClick} to="/">
                Logout
              </Link>
            </div>
          );
        }}
      </MessageContext.Consumer>
    );
  }
}

export default MessageList;