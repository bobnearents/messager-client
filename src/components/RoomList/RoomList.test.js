import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Route, Link, MemoryRouter } from "react-router-dom";

import RoomList from "./RoomList";

describe("<RoomList />", () => {
  it("Renders without crashing", () => {
    shallow(<RoomList />);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <RoomList />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
