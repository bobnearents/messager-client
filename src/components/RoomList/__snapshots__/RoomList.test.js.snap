// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomList /> renders correctly 1`] = `
<div
  className="inactive room-list-container"
>
  <h2>
    Rooms
    <a
      href="/createRoom"
      onClick={[Function]}
    >
      <svg
        className="add-room"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </a>
  </h2>
  <div
    className="room-list"
  />
  <h2>
    Direct Messages
  </h2>
  <div
    className="dm-list"
  />
  <a
    className="logout"
    href="/"
    onClick={[Function]}
  >
    Logout
  </a>
</div>
`;
